test/rest-smoketest.sh

```diff
@@ -58,12 +66,22 @@ mycurl()
     touch ${RESPONSE_CONTENT}
     truncate -s 0 ${RESPONSE_HEADERS}
     truncate -s 0 ${RESPONSE_CONTENT}
-    curl -D ${RESPONSE_HEADERS} \
```

Comment:
rather than having two branches with almost the same code, please build up one array variable of curl options using conditional code and then call curl once with the computed options, e.g. something like:
```
curl_options=(
  -D ${RESPONSE_HEADERS}
  -o ${RESPONSE_CONTENT}
  -s -k
  -w "%{http_code}::%{content_type}::%{size_download}\n"
)

if [[ -n "$GOAUTH" ]]
then
   curl_options+=( -H "Authorization: Globus-Goauthtoken $GOAUTH" )
elif [[ -n "$COOKIES" ]]
then
   curl_options+=( -b "$COOKIES" -c "$COOKIES" )
fi

curl "${curl_options[@]}" "$@"
```
Also, as per this example I would simplify the COOKIES processing to default to no cookies rather than defaulting to `cookie` anymore.